keps/draft-ephemeral-test-environments.md

```diff
@@ -0,0 +1,230 @@
+---
+kep-number: 0
+title: Ephemeral End-to-End Test Environments
+authors:
+  - "@jbarrick-mesosphere"
+  - "@orsenthil"
+owners:
+  - TBD
+  - "@johndoe"
+editor: TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - KEP-0004
+  - KEP-0008
+---
+
+# Ephemeral End-to-End Test Environments
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+   * [Goals](#goals)
+   * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+   * [User Stories](#user-stories)
+   * [Implementation Details](#implementation-details)
+      * [Deploying Terraform test environments](#deploying-terraform-test-environments)
+      * [Maintaining an idle pool of environments](#maintaining-an-idle-pool-of-environments)
+      * [Claiming a test environment](#claiming-a-test-environment)
+      * [Deleting a test environment](#deleting-a-test-environment)
+   * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternative Solutions](#alternative-solutions)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+This document describes how test environments will be provisioned in CI for use in end-to-end testing of KUDO and Frameworks.
+
+A KUDO Framework will be built that can manage idle pools of Terraform-provisioned Kubernetes clusters - ensuring that CI jobs always have Kubernetes available to use for tests without waiting to provision them.
```

Comment:
Are you aware of [boskos](https://github.com/kubernetes/test-infra/tree/master/boskos#background)? From my limited knowledge it seems like it is designed to do the job described in this KEP...
Could we reuse that rather than build something from scratch?

Reply:
Oh this looks great. I actually hadn’t seen this. But, we have our Prow deployment imminent so we’ll be able to try that out and see if it works for us!